.github/workflows/reusable-docs.yml

```
@@ -23,11 +23,13 @@ jobs:
       refspec_pr: '+${{ github.event.pull_request.head.sha }}:remotes/origin/${{ github.event.pull_request.head.ref }}'
     steps:
     - name: 'Check out latest PR branch commit'
+      if: github.event_name == 'pull_request'
       uses: actions/checkout@v4
       with:
         ref: ${{ github.event.pull_request.head.sha }}
```

A cleaner approach is to just make this value conditional, instead duplicating the step. And empty value will trigger the action default behavior.
```suggestion
        ref: >-
          ${{
            github.event_name == 'pull_request'
            && github.event.pull_request.head.sha
            || ''
          }}
```